pkg/sensors/tracing/generickprobe.go

```diff
@@ -728,8 +730,13 @@ func loadMultiKprobeSensor(ids []idtable.EntryID, bpfDir, mapDir string, load *p
 
 		data.Symbols = append(data.Symbols, gk.funcName)
 		data.Cookies = append(data.Cookies, uint64(index))
+
+		if gk.hasOverride && !load.RetProbe {
+			data.Overrides = append(data.Overrides, gk.funcName)
+		}
```

Comment:
The loader code makes my head heart :(

OK, let me see if I got this right:

We have two programs (`load *program.Program`),  one for kprobe entry, and one for kprobe ret (if we have a return argument), and there are two cases: single- and multi-probe.

For multi-probe, If we loading the return program (`load.RetProbe == true`) then override is not relevant, and we  always ignore it. If this is not an overload program, we set the proper `MultiKprobeAttchData` so that the attachment works as expected in `MultiKprobeAttach`.

Does that sound about right?

Would be nice to cleanup the loading code at some point, but I'm not sure how easy would it be. There seems to be some deduplication that can be done between multi- and single- kprobe, but might make sense to also rethink things and figure how to do things properly.

</rant>

Reply:
> The loader code makes my head hurt :(

Reply:
> The loader code makes my head hurt :(

I take my aspirin any time I touch that thing ;-)

yes, override is relevant only for entry probes

it's straightforward for single kprobe where we install the override program directly to that entry probe

for multi kprobe we install the override program only for kprobes that have override action (might be just subset)

and that's what's happening in the code you pointed out.. we gather all 'override' kprobes and install override program to them